catalog/app/components/FileEditor/QuiltConfigEditor/BucketPreferences/State.tsx

```diff
@@ -0,0 +1,200 @@
+  return {
+    'ui.actions.copyPackage': childOfBool(json?.ui?.actions, 'copyPackage'),
+    'ui.actions.createPackage': childOfBool(json?.ui?.actions, 'createPackage'),
+    'ui.actions.deleteRevision': childOfBool(json?.ui?.actions, 'deleteRevision'),
+    'ui.actions.revisePackage': childOfBool(json?.ui?.actions, 'revisePackage'),
+
+    'ui.blocks.analytics': json?.ui?.blocks?.analytics,
+    'ui.blocks.browser': json?.ui?.blocks?.browser,
+    'ui.blocks.code': json?.ui?.blocks?.code,
+
+    'ui.blocks.meta':
+      typeof json?.ui?.blocks?.meta !== 'undefined'
+        ? !!json.ui.blocks.meta
+        : json?.ui?.blocks?.meta,
+    'ui.blocks.meta.user_meta.expanded': isExpanded(json?.ui?.blocks?.meta, 'user_meta'),
+    'ui.blocks.meta.workflows.expanded': isExpanded(json?.ui?.blocks?.meta, 'workflows'),
+
+    'ui.blocks.gallery.files': childOfBool(json?.ui?.blocks?.gallery, 'files'),
+    'ui.blocks.gallery.overview': childOfBool(json?.ui?.blocks?.gallery, 'overview'),
+    'ui.blocks.gallery.packages': childOfBool(json?.ui?.blocks?.gallery, 'packages'),
+    'ui.blocks.gallery.summarize': childOfBool(json?.ui?.blocks?.gallery, 'summarize'),
+
+    'ui.blocks.qurator': json?.ui?.blocks?.qurator,
+
+    'ui.nav.files': childOfBool(json?.ui?.nav, 'files'),
+    'ui.nav.packages': childOfBool(json?.ui?.nav, 'packages'),
+    'ui.nav.queries': childOfBool(json?.ui?.nav, 'queries'),
+
+    'ui.package_description': json?.ui?.package_description,
+    'ui.package_description.multiline': json?.ui?.package_description_multiline,
+
+    'ui.source_buckets': json?.ui?.sourceBuckets && Object.keys(json?.ui?.sourceBuckets),
+    'ui.source_buckets.default': json?.ui?.defaultSourceBucket,
+
+    'ui.athena.defaultWorkgroup': json?.ui?.athena?.defaultWorkgroup,
+  }
+}
+
+type Defaults = Required<ReturnType<typeof parseUser>>
+
+const sys: Defaults = {
+  'ui.actions.copyPackage': true,
+  'ui.actions.createPackage': true,
+  'ui.actions.deleteRevision': false,
+  'ui.actions.revisePackage': true,
+
+  'ui.athena.defaultWorkgroup': '',
+
+  'ui.blocks.analytics': true,
+  'ui.blocks.browser': true,
+  'ui.blocks.code': true,
+
+  'ui.blocks.meta': true,
+  'ui.blocks.meta.user_meta.expanded': false,
+  'ui.blocks.meta.workflows.expanded': false,
+
+  'ui.blocks.gallery.files': true,
+  'ui.blocks.gallery.overview': true,
+  'ui.blocks.gallery.packages': true,
+  'ui.blocks.gallery.summarize': true,
+
+  'ui.blocks.qurator': true,
+
+  'ui.nav.files': true,
+  'ui.nav.packages': true,
+  'ui.nav.queries': true,
+
+  'ui.package_description': {
+    '.*': {
+      message: true as const,
+      user_meta: [] as ReadonlyArray<string>,
+    },
+  },
+  'ui.package_description.multiline': false,
+
+  'ui.source_buckets': [],
+  'ui.source_buckets.default': '',
+}
+
+function val<K extends Key>(
+  key: K,
+  user: Partial<Defaults>,
+  ext: Partial<Defaults>,
+): KeyedValue<K> {
+  return {
+    isDefault: !user[key],
```

Comment:
this means unchecked values are always considered "default" -- is this intended?